lib/trie.js

```diff
@@ -0,0 +1,77 @@
+
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+// insert new nodes to Trie
+Trie.prototype.add = function(route, handler) {
+  var node = this;
+  var parts = getPathParts(route.fullPath);
+  var segment;
+
+  if (!parts.length) {
+    addVerbHandler(node, route, handler);
+    return this;
+  }
+  for (var i = 0; i < parts.length; i++) {
+    segment = parts[i];
+    if (!(segment in node)) {
+      node[segment] = new Trie();
+    }
+    if (i + 1 === parts.length) {
+      addVerbHandler(node[segment], route, handler);
+    }
+    node = node[segment];
+  }
+
+  return this;
+};
+
+// Look for matching path and return methods for a match
+Trie.prototype.find = function(path) {
```

Comment:
IIRC I was commenting earlier that I would like this to become `Trie.prototype.find = function(path, verb)`.